### PR TITLE
Silent Fan profile added (v2.0)

### PR DESCRIPTION
**Pull request by Adravil**
More upgrades to Silent Fan profile.
* SemiSilent Fan profile had been added. It's a copy of the Silent Fan profile that, unlike the Silent Fan profile, allows user to increase the TDP higher than 10W. The SemiSilent Fan profile serves two purposes: 
** firstly as a precausios measure, so that the user will have to manually and intentionally push the TDP (and so the temperature) to its extremum;
** and secondly as simply a notification that to keep the Deck indeed silent there is a Silent Fan profile (10W-, for kind of an "average user"), and the SemiSilent is for those who wants to find a balance by him-/herself (11W+, for kind of an "advanced user").
* Now if user switches to Silent Fan profile in Fan Control or Power Control, while the TDP is set to 11W or higher, the TDP will automatically be pushed down to 10W and the user will be notified.
* Now if user increases TDP to 11W or higher in Power Control, while the Fan profile is set to Silent, the Fan profile will automatically be switched to SemiSilent and the user will be notified.
* Also, 11W, 13W and 14W TDP values were returned back for precise TDP adjustments (for the same SemiSilent Fan profile, for example).

### CODE REVIEW

File: PowerControl/Controller.cs [hunk 1]
```diff
@@ -299,6 +307,35 @@ private Task NeptuneDevice_OnInputReceived(hidapi.HidDeviceInputReceivedEventArg
...
+        private void PowerControlLoopTimer_Tick(object sender, EventArgs e)
+        {
+            if (powerControlLoopTimer is null)
+                return;
+
+            try
+            {
+                powerControlLoopTimer.Enabled = false;
+
+                //if (TDP.Instance != null
```
Comment (Adravil) [inline PowerControl/Controller.cs:319]: System.AccessViolationException: 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.' if I uncomment this. Not familiar with memory management so fixing may take some time.

Comment (Adravil) [reply PowerControl/Controller.cs:319]: Resolved. There's no need in storing the separate "current TDP" value. I can simply switch to 10W TDP in PowerControl if I detect the Silent fan profile being selected (which detection is already implemented in PowerControl and is working fine).

**Comment by ayufan** [comment]
@Adravil This seems like a very complicated piece of code that I believe pushes too much understanding on the user what happens:

- Can we just have a single Silent profile that if system power goes over 10W will have slightly different cure? All, being done from perspective of FanControl.
- I believe it would be enough, since if system will consume more than 11W we could force some 2500-3000RPM minimum.
- In this case the `Silent` will be truly silent on low TDP, but will not less silent on higher power and thus safer.
-

**Comment by ayufan** [comment]
> Also, 11W, 13W and 14W TDP values were returned back for precise TDP adjustments (for the same SemiSilent Fan profile, for example).

Unsure of that. The idea was to to avoid adding those higher TDPs, but might consider this as a separate PR.

**Comment by Adravil** [comment]
> @Adravil This seems like a very complicated piece of code that I believe pushes too much understanding on the user what happens:
> 
>     * Can we just have a single Silent profile that if system power goes over 10W will have slightly different cure? All, being done from perspective of FanControl.
> 
>     * I believe it would be enough, since if system will consume more than 11W we could force some 2500-3000RPM minimum.
> 
>     * In this case the `Silent` will be truly silent on low TDP, but will not less silent on higher power and thus safer.

* I understand your concern, but such change kills my initial idea of having the fan curve, being guided by which Deck's fan would stay silent in 80-90°C temperature range (what, of course, requires user to lower the TDP). So at the end, for me, having the separate Silent fan curve which in the 80-90°C range has the same graph as, for example, the Steam OS fan curve, has minimum to no meaning. I can probably just simply switch to the Steam OS fan curve and have the similar or same result.
* The Silent fan curve already has a minimum RPM set at 3000: _1.28^(x-60)+3000_ [PR 141](https://github.com/ayufan/steam-deck-tools/pull/141).
* I agree that for an average user the Silent fan profile in its current form may be not reliable, so I propose to rollback my previous [PR 141](https://github.com/ayufan/steam-deck-tools/pull/141) from the developer branch (to not mess up future releases) and set this case closed.

**Comment by Adravil** [comment]
> > Also, 11W, 13W and 14W TDP values were returned back for precise TDP adjustments (for the same SemiSilent Fan profile, for example).
> 
> Unsure of that. The idea was to to avoid adding those higher TDPs, but might consider this as a separate PR.

... rollback PR 141 and leave TDP options as they are.

**Comment by Adravil** [comment]
Sorry for the late reply, been busy a bit.

**Comment by ayufan** [comment]
@Adravil 

We could make `Silent` to increase RPM over some actual TDP, but make the threshold configurable via `.ini`, and document this. For you and others aware of risks, people could modify `FanControl.ini` to set `SilentPowerThreshold=15` to move the curve up.

**Comment by Adravil** [comment]
> @Adravil
> 
> We could make `Silent` to increase RPM over some actual TDP, but make the threshold configurable via `.ini`, and document this. For you and others aware of risks, people could modify `FanControl.ini` to set `SilentPowerThreshold=15` to move the curve up, but keep a default conservative of `SilentPowerThreshold=10 or 11`.

That's doable too, tankfully such dependency will be linear:
* at _1.28^(x-60)+3000_ curve reaches 4000 RPM at ~88°C, 5100 RPM at 91°C;
* at _1.28^(x-59)+3000_ curve reaches 4000 RPM at ~87°C, 5100 RPM at 90°C;
* and so on...

To my understanding you'd like this property to not be displayed and configurable through UI to prevent harm, right? If so, I agree, that'd be a good balance.

Working on it.

**Comment by Adravil** [comment]
Closing, preparing v3.0 PR.

**Pull request closed without merging**